### PR TITLE
update and merge to upstream 1.2.8

### PR DESCRIPTION
**Pull request by kingster-will**

### CODE REVIEW

File: pool/store/sqlite_store.py [hunk 1]
```diff
@@ -116,23 +116,56 @@ def _row_to_farmer_record(row) -> FarmerRecord:
         )
 
     async def add_farmer_record(self, farmer_record: FarmerRecord, metadata: RequestMetadata):
```
Comment (LeoHChen) [inline pool/store/sqlite_store.py:118]: I think we need to modify pg_store.py as well as the logic has been updated in sqlite_store.py

Comment (kingster-will) [reply pool/store/sqlite_store.py:118]: I had the same impression at the first beginning. but looked closely at the change in sqlite_store.py, there are no logic changes but a couple of optimizations.
for example,  the changes in `add_farmer_record()`, it just split the one `INSERT or REPLACE` (i.e. upsert) SQL statements into two separated SQL statements, one for insert and another for the update repspectively. 
But in our pg_store.py, we have using the 2 separated SQL statements for the upsert logic in the first place, Hence there is no change is needed in pg_store.py for the merge. 

Another change in the sqlite_store.py is merging two SQL statements into one SQL, but it is purely SQLite SQL syntax, no need to apply to Postgres either.

**Comment by kingster-will** [comment]
Note that, with the upgrade, a new configuration item `store` was introduced.
A new item below required to be added into **config.yaml**, otherwise the pool server could not access the database properly.

```
store: PostgresPoolStore
```

File: pool/store/sqlite_store.py (hunk 1 above)
Comment (kingster-will) [reply pool/store/sqlite_store.py:118]: I have updated pg_store in the last commit with applying "upsert" for adding a new farmer.

**Review by grassr00t — APPROVED** [review]

**Pull request merged**